composition-js/src/__tests__/override.compose.test.ts

```diff
@@ -0,0 +1,270 @@
+import { printSchema } from "@apollo/federation-internals";
+import gql from "graphql-tag";
+import "./matchers";
+import {
+  assertCompositionSuccess,
+  schemas,
+  errors,
+  composeAsFed2Subgraphs,
+} from "./compose.test";
+
+describe("composition involving @override directive", () => {
```

Comment:
We should have a test for putting a `@override` on both a type and one of its field at the same time. Which I believe currently would only use the `@override` on the field for the field having it and would us the `@override` on the type for other fields. And I wonder if that's what we want? I think I'd rather just reject this case with an error for now.